### ISSUE TITLE
hyprland/language sometimes shows nothing

### ISSUE THREAD

**Issue by coffebar** [Alexays/Waybar#3190]
Sometimes I need to change layout to see it in the waybar

![image](https://github.com/Alexays/Waybar/assets/3100053/9c11ca38-fd67-45ce-8813-630ccbdbbf45)

like this:

![image](https://github.com/Alexays/Waybar/assets/3100053/445dbb7d-a052-46ce-98ec-cb7323df58bb)

```
"hyprland/language": {
    "format": "{}",
    "max-length": 18
},
```

I have 2 monitors and 2 keyboards. Also using [hyprland-per-window-layout ](https://github.com/coffebar/hyprland-per-window-layout), not sure if it can be related. 

#### hyprland config:
```
input {
  kb_layout = us,ua
  kb_variant =
  kb_model =
  kb_options = grp:alt_shift_toggle,caps:backspace
  kb_rules =
...
```

pls help me to debug this

**Comment by KalaayPT** [comment]
whats your log if you run waybar from terminal?
also, i had the issue where it wouldnt appear at all, and apparently it was because it couldnt connect to the ipc, giving me this:
`Hyprland IPC: Couldn't connect to /tmp/hypr/70ebc3add3b941a52f460f02129ef0842f580ec7_1714744275_1686582762.socket.sock.`
what i gather from this is that it was apparently looking in the wrong place from some reason, because this
`ln -s $XDG_RUNTIME_DIR/hypr /tmp/hypr`
completely fixed the issue. thought id put this here so this can get fixed/help others

**Comment by coffebar** [comment]
I don't have any errors in my log, just events like this:

> [2024-05-06 18:53:48.588] [debug] hyprland IPC received workspace>>2
> [2024-05-06 18:53:48.589] [debug] hyprland IPC received activewindowv2>>56baeb1bcae0
> [2024-05-06 18:53:48.591] [debug] hyprland IPC received workspacev2>>2,2
> [2024-05-06 18:53:49.734] [debug] hyprland IPC received activewindow>>Alacritty,waybar -l debug
> [2024-05-06 18:53:49.737] [debug] hyprland IPC received activewindowv2>>56baecfa3db0
> [2024-05-06 18:54:45.493] [debug] hyprland IPC received activewindow>>Alacritty:~
> [2024-05-06 18:54:45.501] [debug] hyprland IPC received activewindowv2>>56baec2a33a0
> [2024-05-06 18:54:46.393] [debug] hyprland IPC received activewindow>>Alacritty,waybar -l debug
> [2024-05-06 18:54:46.400] [debug] hyprland IPC received activewindowv2>>56baecfa3db0

**Comment by coffebar** [comment]
Did find interesting: 
> [2024-05-06 18:51:55.642] [debug] hyprland language onevent with English (US)
> [2024-05-06 18:51:55.648] [debug] hyprland language onevent with English (US)
> [2024-05-06 18:51:55.649] [debug] hyprland IPC received activelayout>>wlr_virtual_keyboard_v1,English (US)
> [2024-05-06 18:51:55.655] [debug] hyprland language onevent with English (US)
> [2024-05-06 18:51:55.661] [debug] hyprland language onevent with English (US)
> [2024-05-06 18:51:55.662] [debug] hyprland IPC received activelayout>>wlr_virtual_keyboard_v1,error
> [2024-05-06 18:51:55.667] [debug] hyprland language didn't find matching layout
> [2024-05-06 18:51:55.667] [debug] hyprland language onevent with error
> [2024-05-06 18:51:55.675] [debug] hyprland language didn't find matching layout
> [2024-05-06 18:51:55.675] [debug] hyprland language onevent with error

**Comment by zjeffer** [comment]
> whats your log if you run waybar from terminal? also, i had the issue where it wouldnt appear at all, and apparently it was because it couldnt connect to the ipc, giving me this: `Hyprland IPC: Couldn't connect to /tmp/hypr/70ebc3add3b941a52f460f02129ef0842f580ec7_1714744275_1686582762.socket.sock.` what i gather from this is that it was apparently looking in the wrong place from some reason, because this `ln -s $XDG_RUNTIME_DIR/hypr /tmp/hypr` completely fixed the issue. thought id put this here so this can get fixed/help others
> 
> EDIT: my IPC was active of course, because `echo $HYPRLAND_INSTANCE_SIGNATURE` did return `70ebc3add3b941a52f460f02129ef0842f580ec7_1714744275_1686582762` which is exactly the thing it was looking for in /tmp/hypr

That has been fixed in #3183, and now also in the new 0.10.3 release of Waybar. The problem was a breaking change by Hyprland.
